standalone-metastore/metastore-server/src/test/java/org/apache/hadoop/hive/metastore/TestPartitionManagement.java

```diff
@@ -654,6 +656,42 @@ public void testNoPartitionRetentionForReplTarget() throws TException, Interrupt
     assertEquals(3, partitions.size());
   }
 
+  @Test
+  public void testPartitionExprFilter() throws TException, IOException {
+    String dbName = "db10";
+    String tableName = "tbl10";
+    Map<String, Column> colMap = buildAllColumns();
+    List<String> partKeys = Lists.newArrayList("state", "dt");
```

Comment:
can you a testcase for timestamp too? timestamp is always the problematic type in partition column.

Reply:
sure, will add timestamp test case as well.